sdk_extensions/zpages/src/test/java/io/opentelemetry/sdk/extensions/zpages/TracezZPageHandlerTest.java

```diff
@@ -0,0 +1,289 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.extensions.zpages;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import io.opentelemetry.sdk.internal.TestClock;
+import io.opentelemetry.sdk.trace.TracerSdkProvider;
+import io.opentelemetry.trace.EndSpanOptions;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.Status.CanonicalCode;
+import io.opentelemetry.trace.Tracer;
+import java.io.ByteArrayOutputStream;
+import java.io.OutputStream;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.mockito.junit.MockitoRule;
+
+/** Unit tests for {@link TracezZPageHandler}. */
+@RunWith(MockitoJUnitRunner.class)
+public final class TracezZPageHandlerTest {
+  private static final String FINISHED_SPAN_ONE = "FinishedSpanOne";
+  private static final String FINISHED_SPAN_TWO = "FinishedSpanTwo";
+  private static final String RUNNING_SPAN = "RunningSpan";
+  private static final String LATENCY_SPAN = "LatencySpan";
+  private static final String ERROR_SPAN = "ErrorSpan";
+  private final TestClock testClock = TestClock.create();
+  private final TracerSdkProvider tracerSdkProvider =
+      TracerSdkProvider.builder().setClock(testClock).build();
+  private final Tracer tracer = tracerSdkProvider.get("TracezZPageHandlerTest");
+  private final TracezSpanProcessor spanProcessor = TracezSpanProcessor.newBuilder().build();
+  private final TracezDataAggregator dataAggregator = new TracezDataAggregator(spanProcessor);
+  @Mock Map<String, String> queryMap;
```

Comment:
In general, you should think of mockito as a hack, a very important hack, but still one that uses bytecode manipulation to circumvent normal Java for testing (now adays it can even mock out final classes, wow). For normal containers even in tests it's best to avoid that (can just pass `ImmutableMap.of()` without much loss in expressiveness).

Don't forget that this now ties down the implementation - `.isEmpty` returns `false` for this map even though you're trying to test the empty map case, so it's not correct and we're just sort of lucky the test passes.

Reply:
Thanks for the explanation! Seems like I don't need mockito here then, should I change the runner back to `JUnit4` runner?

Reply:
Sure

Reply:
Ah just noticed - either way, you definitely shouldn't ever have both the runner and the rule but looks like you can remove both

Reply:
So if in case I need to use @Mock, only mockito rule with JUni4 runner is sufficient?

Reply:
Yeah either mockito rule + junit4 runner or mockito runner, but not both. Which one is a matter of personal preference.

Reply:
Got it, I got confused by one of the previous comment regarding this.